pkg/operator/status.go

```diff
@@ -607,6 +607,14 @@ func isMachineConfigPoolConfigurationValid(pool *mcfgv1.MachineConfigPool, versi
 	if renderedMC.Spec.OSImageURL != osURL {
 		return fmt.Errorf("osImageURL mismatch for %s in %s expected: %s got: %s", pool.GetName(), renderedMC.Name, osURL, renderedMC.Spec.OSImageURL)
 	}
+
+	// check that the rendered config matches the OCP release version for cases where there is no OSImageURL change nor no new MCO commit
+	if !inClusterBringup {
+		rv, ok := renderedMC.Annotations[ctrlcommon.ReleaseImageVersionAnnotationKey]
+		if ok && rv != releaseVersion {
```

Comment:
Here we'll silently pass if the annotation doesn't exist.  But, can we instead actually just require that the annotation exist?  

Not requiring it is definitely a more conservative choice.  I am just trying to think through whether it also means that fixing this bug requires a new version of the operator as the upgrade *source* and not just the upgrade target.

May be at least worth a log message emitted if it doesn't exist.

Reply:
agree on adding the log :+1:

Reply:
did some testing and ended up adding a !ok check. didn't see any issues after doing some testing, so let's do it.